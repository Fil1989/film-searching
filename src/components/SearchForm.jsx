import { useState } from 'react'
import TextField from '@mui/material/TextField'
import Button from '@mui/material/Button'
import films from '../data/imdb.json'
import amountControles from '../controles/amountControles.js'
const SearchForm = ({ setResults, setVisible }) => {
  const [search, setSearch] = useState('')

  return (
    <form
      className="search_form"
      onSubmit={e => {
        e.preventDefault()
        amountControles.resetPage()
        const resultsOfSearch = films.filter(
          film =>
            film.title &&
            film.title
              .toLowerCase()
              .includes(
                e.currentTarget[0].attributes[4].nodeValue.toLowerCase(),
              ),
        )
        amountControles.setAllResults = resultsOfSearch
        setResults(amountControles.getResults)
        setTimeout(() => {
          setVisible(true)
        }, 500)
      }}
    >
      <TextField
        sx={{
          marginRight: '20px',
        }}
        value={search}
        onChange={e => {
          e.preventDefault()
          setSearch(e.target.value)
        }}
        label="Search"
        type="text"
        variant="outlined"
        name="search"
        className="search_input"
      />
      <Button type="submit" variant="contained">
        Search
      </Button>
    </form>
  )
}
export default SearchForm
