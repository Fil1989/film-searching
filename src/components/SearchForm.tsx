import { useState } from 'react'
import TextField from '@mui/material/TextField'
import Button from '@mui/material/Button'
import films from '../data/imdb.json'
import amountControles from '../controles/amountControles'
import { IResults } from '../interfaces/main.interface'

interface IProps {
  setResults: (films: IResults[]) => void
  setVisible: (value: boolean) => void
}
export const value: string = '778'

const SearchForm = ({ setResults, setVisible }: IProps) => {
  const [search, setSearch] = useState<string>('')

  return (
    <form
      className="search_form"
      onSubmit={e => {
        e.preventDefault()
        amountControles.resetPage()
        const resultsOfSearch = films.filter(
          film =>
            film.title &&
            search &&
            film.title.toLowerCase().includes(search.toLowerCase()),
        )
        amountControles.setAllResults = resultsOfSearch
        setResults(amountControles.getResults)
        // setTimeout(() => {
        setVisible(true)
        // }, 500)
      }}
    >
      <TextField
        sx={{
          marginRight: '20px',
        }}
        value={search}
        onChange={e => {
          setSearch(e.target.value)
        }}
        label="Search"
        type="text"
        variant="outlined"
        name="search"
        className="search_input"
        inputProps={{ 'data-testid': 'search_field' }}
      />
      <Button type="submit" variant="contained" data-testid="searching_btn">
        Search
      </Button>
      <p data-testid="value">{value}</p>
    </form>
  )
}
export default SearchForm
// export const sum = (a: number = 0, b: number = 0): number => a + b
