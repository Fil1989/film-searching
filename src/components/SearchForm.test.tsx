import { render, fireEvent } from '@testing-library/react'
import SearchForm /*, { sum }*/ from './SearchForm'
// import WrongWay from './WrongWay.tsx'
// import  from './SearchForm.tsx'

// describe('sum', () => {
//   it('2+2', () => {
//     expect(sum(2, 2)).toBe(4)
//   })
//   it('2+10', () => {
//     expect(sum(2, 10)).toBe(12)
//   })
//   it('20+10', () => {
//     expect(sum(20, 10)).toBe(30)
//   })
//   it('NaN check', () => {
//     expect(sum(20)).not.toBeNaN()
//     expect(sum()).not.toBeNaN()
//   })
// })
describe('SearchForm', () => {
  it('snapshot test', () => {
    const { container } = render(
      <SearchForm setResults={() => {}} setVisible={() => {}} />,
    )
    expect(container).toMatchSnapshot()
  })
  it('getByTestId test', () => {
    const { getByTestId } = render(
      <SearchForm setResults={() => {}} setVisible={() => {}} />,
    )
    expect(getByTestId('value').textContent).toBe('778')
  })
  it('search initializing test', () => {
    const Searching = jest.fn()
    const { getByTestId } = render(
      <SearchForm setResults={Searching} setVisible={() => {}} />,
    )
    getByTestId('searching_btn').click()
    getByTestId('searching_btn').click()
    getByTestId('searching_btn').click()
    expect(Searching).toBeCalled()
    expect(Searching).toBeCalledTimes(3)
  })
  it('intersection observer btn test', () => {
    const IntersectionBtn = jest.fn()
    const { getByTestId } = render(
      <SearchForm setResults={() => {}} setVisible={IntersectionBtn} />,
    )
    getByTestId('searching_btn').click()
    expect(IntersectionBtn).toBeCalled()
  })
  it('input correctness test', () => {
    const utils = render(
      <SearchForm setResults={() => {}} setVisible={() => {}} />,
    )
    const input = utils.getByTestId('search_field') as HTMLInputElement
    fireEvent.change(input, {
      target: { value: '23456' },
    })
    expect(input.value).toBe('23456')
  })
})
