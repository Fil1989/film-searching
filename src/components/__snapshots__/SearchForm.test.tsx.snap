// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm snapshot test 1`] = `
<div>
  <form
    class="search_form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root search_input css-l7l03d-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
      >
        Search
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          data-testid="search_field"
          name="search"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Search
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="searching_btn"
      tabindex="0"
      type="submit"
    >
      Search
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
